src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1685,6 +1695,36 @@ else if (environmentName != null) {
         return changesMade;
     }
 
+    /**
+     * Return the environment change list
+     *
+     * @param preExistingEnvs
+     *  Pre existing environment list
+     * @param updatedEnvs
+     *  Updated environment list
+     *
+     * @return
+     *  Returns the unique environmentIds that are added, removed or reordered
+     */
+    private List<String> getEnvironmentChangeList(List<String> preExistingEnvs, List<String> updatedEnvs) {
```

Comment:
This method is conceptually also playing a role in the filtering of entitlements, so it could be rolled into the new `EntitlementEnvironmentFilter` class. The whole interface could be used with a fluent API as such:
```
new EntitlementEnvironmentFIlter()
    .setPreExistingEnvironments(preExistingEnvs)
    .setUpdatedEnvironments(updatedEnvs)
    .setConsumer(toUpdate)
    .filterEntitlements();
```

Reply:
Done